internal/transport/http_util.go

```diff
@@ -120,8 +118,32 @@ type decodeState struct {
 	statsTags      []byte
 	statsTrace     []byte
 	contentSubtype string
+}
+
+func newDecodeState(serverSide bool, ignoreContentType bool) *decodeState {
+	return &decodeState{
+		serverSide:        serverSide,
+		ignoreContentType: ignoreContentType,
+	}
+}
+
+// Records the states during HPACK decoding. Must be reset once the
```

Comment:
This comment applies more for parsedHeaderData than decodeState, doesn't it?

Reply:
moved comments.